### ISSUE TITLE
Can't get the reported results

### ISSUE THREAD

**Issue by gawainx** [migalkin/ilpc2022#7]
I clone the repo and run the suggesting command `python main.py --dataset small -d 32 -e 50 -n 16 -m 5.0 -lr 0.0001` to obtain results on small dataset. However, the results are confusing and very low, far away from the reported results.

Results on my exp are:
```
INFO:pykeen.evaluation.evaluator:Evaluation took 0.15s seconds
INFO:root:Test MRR       : 0.00233
INFO:root:Test Hits@100  : 0.02050
INFO:root:Test Hits@10   : 0.00396
INFO:root:Test Hits@5    : 0.00224
INFO:root:Test Hits@3    : 0.00086
INFO:root:Test Hits@1    : 0.00069
INFO:root:Test AMRI      : -0.00855
```

My environments:

- PyTorch 1.11
- PyKeen 1.90
- Tesla V100 32GB
- Ubuntu 18.04
- Random-seed 42 (in the code setting with `set_random_seed(42)`)

Any suggestions about how to solve this problem?

Thx !

**Comment by mberr** [comment]
HI @gawainx ,

could you try re-running to code with `pykeen~=1.8.0` installed and report the results? If I remember correctly, this was the version we ran the experiments reported in the README with.

**Comment by gawainx** [comment]
> HI @gawainx ,
> 
> could you try re-running to code with `pykeen~=1.8.0` installed and report the results? If I remember correctly, this was the version we ran the experiments reported in the README with.

Thank you for your reply ! I reinstall PyKeen with version 1.8.0 and re-run the code. Results are very close to the reported results:

```
INFO:pykeen.evaluation.evaluator:Evaluation took 0.14s seconds
INFO:root:Test MRR       : 0.03457
INFO:root:Test Hits@100  : 0.47743
INFO:root:Test Hits@10   : 0.08856
INFO:root:Test Hits@5    : 0.04290
INFO:root:Test Hits@3    : 0.01757
INFO:root:Test Hits@1    : 0.00345
INFO:root:Test AMRI      : 0.68331
```

Also, I notice that the loss of coverage model is very different from 1.9.0 version:

- PyKeen 1.8.0 - `Metric: loss = 0.00031721685887280734`
- PyKeen 1.9.0 - `Metric: loss = 0.52316818891502`

So, I guess the reason of strange results is that the implementation of `NSSALoss(margin=margin)` or Negative Sampling mechanism in 1.9.0.

**Comment by mberr** [comment]
Regarding the loss, there was https://github.com/pykeen/pykeen/pull/1021, which fixed https://github.com/pykeen/pykeen/issues/1020 - this was however only affecting the *epoch* loss (which had been normalized incorrectly), but not the batch-level losses used for calculating the gradients.

However, it looks like something changed :sweat_smile: I tried running the same script for 1.8.2. and still get the original results;
```
INFO:pykeen.evaluation.evaluator:Evaluation took 0.16s seconds
INFO:root:Test MRR       : 0.03457
INFO:root:Test Hits@100  : 0.47519
INFO:root:Test Hits@10   : 0.08908
INFO:root:Test Hits@5    : 0.04290
INFO:root:Test Hits@3    : 0.01792
INFO:root:Test Hits@1    : 0.00379
INFO:root:Test AMRI      : 0.68272
```
With 1.9.0 it is however much lower, as you already observed. So the difference must lie somewhere in https://github.com/pykeen/pykeen/compare/v1.8.2...v1.9.0.

For now, I would suggest to use PyKEEN 1.8.2. until we found the source of the difference.

**Comment by mberr** [comment]
For now, I would suggest to use PyKEEN 1.8.2. until we found the source of the difference.

**Comment by gawainx** [comment]
> Regarding the loss, there was [pykeen/pykeen#1021](https://github.com/pykeen/pykeen/pull/1021), which fixed [pykeen/pykeen#1020](https://github.com/pykeen/pykeen/issues/1020) - this was however only affecting the _epoch_ loss (which had been normalized incorrectly), but not the batch-level losses used for calculating the gradients.
> 
> However, it looks like something changed 😅 I tried running the same script for 1.8.2. and still get the original results;
> 
> ```
> INFO:pykeen.evaluation.evaluator:Evaluation took 0.16s seconds
> INFO:root:Test MRR       : 0.03457
> INFO:root:Test Hits@100  : 0.47519
> INFO:root:Test Hits@10   : 0.08908
> INFO:root:Test Hits@5    : 0.04290
> INFO:root:Test Hits@3    : 0.01792
> INFO:root:Test Hits@1    : 0.00379
> INFO:root:Test AMRI      : 0.68272
> ```
> 
> With 1.9.0 it is however much lower, as you already observed. So the difference must lie somewhere in [pykeen/pykeen@v1.8.2...v1.9.0](https://github.com/pykeen/pykeen/compare/v1.8.2...v1.9.0).
> 
> For now, I would suggest to use PyKEEN 1.8.2. until we found the source of the difference.

I notice that in 1.9.0 version, the `NodePieceRepresentation` class has new `combination` args, related to new file `pykeen.nn.combination` and class `CombinedRepresentation`, while the 1.8.2 version of `NodePieceRepresentation` implement the forward logic by `self.aggregation` (refer to pykeen/nn/node_piece/representation.py, line 311). 

Hope this observation could help you locate the bugs 🐛

Anyway, now I install the 1.8.2 version and obtain similar performances. Thx again. 👍

**Comment by mberr** [comment]
Hi @gawainx ,

the error should be fixed now in PyKEEN's `master` branch (with https://github.com/pykeen/pykeen/pull/1104 being merged). Until our next release, you would need to install it directly from GitHub via
```shell
pip install git+https://github.com/pykeen/pykeen.git
```
Of course, you can also keep on using version 1.8.2.

Thanks for finding the issue! 👍
